ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java

```diff
@@ -3612,8 +3612,8 @@ private void testRenameTable(boolean blocking) throws Exception {
       txnMgr2.acquireLocks(driver2.getPlan(), ctx, null, false);
       locks = getLocks();
 
-      ShowLocksResponseElement checkLock = checkLock(LockType.EXCLUSIVE,
-        LockState.WAITING, "default", "tab_acid", null, locks);
+      ShowLocksResponseElement checkLock = checkLock(LockType.SHARED_READ,
```

Comment:
Why we don't need an exclusive lock during a rename?

* what if we are reading data from a table in some dag
* in the meantime someone renames the table

with `SHARED_READ` the table will be renamed instantly - which will cause the dag to fail; or not?

Reply:
this is the most serious issue - without addressing this I'm against this change